frontend/src/app/Nav.jsx

```diff
@@ -1,6 +1,187 @@
+    onFirstModalOpen();
+  };
+
+  const updateAllOfflineNotifs = async (req, res) => {
+    try {
+      const response = await axios.put(
+        `${BACKEND_URL}/notif/update-all-offline-notifications`,
+        {},
+        {
+          headers: {
+            Authorization: `Bearer ${token}`,
+          },
+          withCredentials: true,
+        }
+      );
+    } catch (error) {
+      setError("Error updating notification");
+    } finally {
+      getNotifications();
+      onSecondModalClose();
+    }
+  };
+
+  const updateReadNotifs = async (notifId) => {
+    try {
+      const response = await axios.put(
+        `${BACKEND_URL}/notif/${notifId}/update-read-notifications`,
+        {},
+        {
+          headers: {
+            Authorization: `Bearer ${token}`,
+          },
+          withCredentials: true,
+        }
+      );
+    } catch (error) {
+      setError("Error updating notification");
+    }
+  };
+
+  const goToFile = async (notifId, groupId, fileId) => {
+    navigate(`/group/${groupId}/files/${fileId}/workstation`);
+    updateReadNotifs(notifId);
+    updateAllOfflineNotifs();
+    getNotifications();
+  };
+  const goToCodeGroup = (notifId, groupId) => {
+    navigate(`/group/${groupId}`);
+    updateReadNotifs(notifId);
+    updateAllOfflineNotifs();
+    getNotifications();
+  };
+
+  useEffect(() => {
+    if (offlineNotifications.length > 0) {
+      openWelcomeBackModal();
+    }
+  }, [notifications]);
+
+  const {
+    isOpen: isFirstModalOpen,
+    onOpen: onFirstModalOpen,
+    onClose: onFirstModalClose,
+  } = useDisclosure();
+  const {
+    isOpen: isSecondModalOpen,
+    onOpen: onSecondModalOpen,
+    onClose: onSecondModalClose,
+  } = useDisclosure();
+
+  return (
+    <div id="Nav">
+      <i className="fa-solid fa-bell" onClick={onSecondModalOpen}></i>
+
+      <Modal isOpen={isFirstModalOpen} onClose={onFirstModalClose}>
+        <ModalOverlay />
+        <ModalContent>
+          <ModalHeader> Welcome back</ModalHeader>
+          <ModalCloseButton />
+          <ModalBody>
+            <Box>You had some notifications while you were away.</Box>
+          </ModalBody>
+
+          <ModalFooter>
+            <Button
+              colorScheme="blue"
+              mr={3}
+              onClick={() => {
+                onFirstModalClose();
+                onSecondModalOpen();
+              }}
+            >
+              View All Now
+            </Button>
+            <Button onClick={onFirstModalClose} variant="outline">
+              Close
+            </Button>
+          </ModalFooter>
+        </ModalContent>
+      </Modal>
+
+      <Modal isOpen={isSecondModalOpen} onClose={onSecondModalClose}>
+        <ModalOverlay />
+        <ModalContent>
+          <ModalHeader>Notifications</ModalHeader>
+          <ModalCloseButton />
+          <ModalBody>
+            <div>
+              <div id="notifs">
+                {notifications.length > 0 &&
```

Comment:
Since this popup is meant to highlight offline notifications, I think you can only show offline notifs here, and not all notifs. And then you can remove the 'Offline notifications shown in gold' text too

Reply:
I can do this, but the idea is for a user to be able to click on the bell icon to see all notification at any point in the app. The color distinction just separates new offline notifications from regular ones. Should I still update it?

Reply:
Isn't that what the Notifications pane is for? To see all the notifications. While this modal is to see the offline missed notifications.